stdlib/result.gr

```diff
@@ -1,89 +1,200 @@
+ *
+ * @param result: The result to convert
+ * @returns `Some(value)` if the Result is `Ok(value)` or `None` if the Result is an `Err`
+ */
+export let toOption = result => {
+  match (result) {
+    Ok(x) => Some(x),
+    _ => None,
+  }
 }
 
-export let flatMap = (fn, v) => {
-    match (v) {
-        Ok(x)  => fn(x),
-        Err(e) => Err(e)
-    }
+/**
+ * If the Result is `Ok(value)`, applies the given function to the `value` to produce a new Result.
+ *
+ * @param fn: The function to call on the value of an `Ok` variant
+ * @param result: The result to map
+ * @returns A new Result produced by the mapping function if the variant was `Ok` or the unmodified `Err` otherwise
+ */
+export let flatMap = (fn, result) => {
+  match (result) {
+    Ok(x) => fn(x),
+    Err(e) => Err(e),
+  }
 }
 
-export let flatMapErr = (fn, v) => {
-    match (v) {
-        Err(e) => fn(e),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is an `Err(value)`, applies the given function to the `value` to produce a new Result.
+ *
+ * @param fn: The function to call on the value of an `Err` variant
+ * @param result: The result to map
+ * @returns A new Result produced by the mapping function if the variant was `Err` or the unmodified `Ok` otherwise
+ */
+export let flatMapErr = (fn, result) => {
+  match (result) {
+    Err(e) => fn(e),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let map = (fn, v) => {
-    match (v) {
-        Ok(x)  => Ok(fn(x)),
-        Err(e) => Err(e)
-    }
+/**
+ * If the Result is `Ok(value)`, applies the given function to the `value` and wraps the new value in an `Ok` variant.
+ *
+ * @param fn: The function to call on the value of an `Ok` variant
+ * @param result: The result to map
+ * @returns A new `Ok` variant produced by the mapping function if the variant was `Ok` or the unmodified `Err` otherwise
+ */
+export let map = (fn, result) => {
+  match (result) {
+    Ok(x) => Ok(fn(x)),
+    Err(e) => Err(e),
+  }
 }
 
-export let mapErr = (fn, v) => {
-    match (v) {
-        Err(e) => Err(fn(e)),
-        Ok(x)  => Ok(x)
-    }
+/**
+ * If the Result is `Err(value)`, applies the given function to the `value` and wraps the new value in an `Err` variant.
+ *
+ * @param fn: The function to call on the value of an `Err` variant
+ * @param result: The result to map
+ * @returns A new `Err` variant produced by the mapping function if the variant was `Err` or the unmodified `Ok` otherwise
+ */
+export let mapErr = (fn, result) => {
+  match (result) {
+    Err(e) => Err(fn(e)),
+    Ok(x) => Ok(x),
+  }
 }
 
-export let mapWithDefault = (fn, def, v) => {
-    match (v) {
-        Ok(a) => fn(a),
-        _     => def
-    }
+/**
+ * If the Result is `Ok(value)`, applies the given function to the `value` to produce a new value, otherwise uses the default value. Useful for unwrapping a successful Result while providing a fallback for any errors that can occur.
```

Comment:
```suggestion
 * If the Result is `Ok(value)`, applies the given function to the `value` to produce a new value, otherwise uses the default value.
 * Useful for unwrapping a successful Result while providing a fallback for any errors that can occur.
```
Let's put on multiple lines.